src/main/scala/strawman/collection/View.scala

```diff
@@ -286,11 +272,11 @@ object IndexedView {
     def apply(i: Int) = underlying.apply(i)
   }
 
-  class TakeRight[A](underlying: IndexedView[A], n: Int)
-  extends View.TakeRight(underlying, n) with IndexedView[A] {
-    override def iterator() = super.iterator() // needed to avoid "conflicting overrides" error
+  class TakeRight[A](underlying: IndexedView[A], n: Int) extends IndexedView[A] {
+    private val normN = n max 0
```

Comment:
You don't need to store this value.  `def length = underlying.length - delta` is enough.